### DISCUSSION TITLE
3 switch with physical touch buttons using TTP223

### DISCUSSION THREAD

**Discussion by MarkDEGT** [HomeSpan/HomeSpan#886] [Q&A]
Thanks a lot for the HomeSpan project.  I´m newbie about Arduino programming.  I´ve spend several days trying to figure out how to add physical buttons to control 3 lamps, here in Central America is very common the use of 3 lamp switches in the wall.  So here are the files I´ve gathered together with the help of ChatGPT and they are working just right with the use of touch sensors TTP223. 

[Concina_sw_2024_HK.txt](https://github.com/user-attachments/files/16203399/Concina_sw_2024_HK.txt)
[DEV_LED.txt](https://github.com/user-attachments/files/16203400/DEV_LED.txt)

Hope them be of some help for newbies like me. 

Kind regards.

**Comment by anthonymanoel** [comment]
Hola Mark!

Mark, how do you did the connections of your physical pushbuttons on ESP?

Here in Brazil, we normally use 2 physical buttons so i will change 1 for Homekit but didn't find some way to do this yet... I was seeing TXT uploaded and is pretty good but i have this question yet. I've tried using relee as intermediator between physical and esp but didn't work...

Anyway i was thinking in use pulsebuttons of Legrand in my home...

**Reply by MkFig** [reply]
Hola Anthony, i´m using a touch sensor the TTP223, you can configure this one with a high output or low output when touch sensing.  With the aforementioned code I am using 3 TTP223, one for each light I want to control.  In the [Concina_sw_2024_HK.txt](https://github.com/user-attachments/files/16203399/Concina_sw_2024_HK.txt)  are 3 inputs: 
new DEV_SimpleLED(17, 27);   // Pins: LED, Power Button for light 1, 
new DEV_SimpleLED(19, 25);   // Pins: LED, Power Button for light 2, 
new DEV_SimpleLED(21, 32);   // Pins: LED, Power Button for light 3.

for each case, LED pin is an output to control your lights ( pin 17, 19, 21) and the Power Button pins are inputs that receiving the signal for each TTP223.  When the signal change from low to high the output (LED PIN) change it state. 

If you check the DEV_LEV.txt  you will found the following line: 

 pinMode(powerPin, INPUT_PULLUP);  // Habilita la resistencia de pull-up interna

That activates the pull-up resistor in  every Power Button pin to avoid false triggering.  

You can use this code with 1, 2, 3, or any number of laps while you have pins available in your esp32.  Just copy the code. 

To control the actual AC lamp, in your case in Brazil I remember is 220V, the LED pin is used as input signal to an relé module to active your lights.  I´m using an Triac module because rele modules since low cost and chineesse are low quality and fails in 3 o 4 months of normal use. 

If you need the functionality Know as 3-way switch, just copy the same PIN number to another light and use that pin as the 3-way switch. 

You just need a low pulse to active the switch.   Hope this helps to solve your question. 

Regards.
